IOMC/RandomEngine/src/RandomNumberGeneratorService.cc

```diff
@@ -215,11 +215,11 @@ namespace edm {
       }
       unsigned int moduleID = mcc->moduleDescription()->id();
 
-      std::vector<ModuleIDToEngine> const& moduleIDVector = streamModuleIDToEngine_.at(streamID.value());
+      std::vector<ModuleIDToEngine>& moduleIDVector = streamModuleIDToEngine_.at(streamID.value());
```

Comment:
@Dr15Jones Yes, it was caused by the ModuleIDToEngine needing to return a non-const random number generator engine.